settings.h

```diff
@@ -10,20 +10,74 @@
    do not use this unless you are certain your application won't use SSE */
 /* #define LIBCO_NO_SSE */
 
-#if defined(LIBCO_C)
-  #if defined(LIBCO_MP)
-    #define thread_local __thread
-  #else
+#if !defined(thread_local) // User can override thread_local for obscure compilers
+  #if !defined(LIBCO_MP) // Running in single-threaded environment
     #define thread_local
+  #else // Running in multi-threaded environment
+    #if defined(__STDC_VERSION__) // Compiling as C Language
+      #if defined(_MSC_VER) // Don't rely on MSVC's C11 support
+        #define thread_local __declspec(thread)
+      #elif __STDC_VERSION__ < 201112L // If we are on C90/99
+        #if defined(__clang__) || defined(__GNUC__) // Clang and GCC
+          #define thread_local __thread
+        #else // // Otherwise, we ignore the directive (unless user provides their own)
```

Comment:
Is this double-comment (`// //`) deliberate for alignment, or a mistake?

Reply:
Oh, it's just a copy-paste mistake.